src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -57,8 +63,18 @@ public EditCommand parse(String args) throws ParseException {
         if (argMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
             editPersonDescriptor.setAddress(ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get()));
         }
-        if (argMultimap.getValue(PREFIX_USERTYPE).isPresent()) {
-            editPersonDescriptor.setUserType(ParserUtil.parseUserType(argMultimap.getValue(PREFIX_USERTYPE).get()));
+        // set the User as a Buyer if he/she has a preference
+        if (argMultimap.getValue(PREFIX_PREFERENCE).isPresent()) {
+            editPersonDescriptor.setUserType(UserTypeUtil.createBuyer());
+            editPersonDescriptor.setPreference(ParserUtil.parsePreference(
+                    argMultimap.getValue(PREFIX_PREFERENCE).get()));
+            editPersonDescriptor.clearProperties();
+        }
+        // set the User as a Seller if he/she has a property/properties
+        if (argMultimap.getValue(PREFIX_PROPERTY).isPresent()) {
+            System.out.print(": " + argMultimap.getValue(PREFIX_PROPERTY));
+            editPersonDescriptor.setUserType(UserTypeUtil.createSeller());
+            editPersonDescriptor.clearPreference();
         }
         parsePropertiesForEdit(argMultimap.getAllValues(PREFIX_PROPERTY)).ifPresent(
                 editPersonDescriptor::setProperties);
```

Comment:
Should this be executed only if `argMultimap.getValue(PREFIX_PROPERTY).isPresent()` is true?